pkg/discovery/network/weavenet.go

```diff
@@ -34,16 +34,22 @@ func discoverWeaveNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, erro
 	var clusterNetwork *ClusterNetwork
 
 	for i := range weaveNetPod.Spec.Containers {
+		var found bool
 		for _, envVar := range weaveNetPod.Spec.Containers[i].Env {
 			if envVar.Name == "IPALLOC_RANGE" {
 				clusterNetwork = &ClusterNetwork{
 					PodCIDRs:      []string{envVar.Value},
 					NetworkPlugin: cni.WeaveNet,
 				}
-
+				found = true
 				break
 			}
 		}
+		if !found {
+			clusterNetwork = &ClusterNetwork{
+				NetworkPlugin: cni.WeaveNet,
+			}
+		}
```

Comment:
Since we assume the CNI is Weave if any Weave pod is present, we can simplify this slightly:

```
		clusterNetwork = &ClusterNetwork{
			NetworkPlugin: cni.WeaveNet,
		}
		for _, envVar := range weaveNetPod.Spec.Containers[i].Env {
			if envVar.Name == "IPALLOC_RANGE" {
				clusterNetwork.PodCIDRs = []string{envVar.Value}

				break
			}
		}
```

Reply:
😆